### PR TITLE
Refs #22165 - Add installer support for disabling hsts

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: templates/settings.yaml.erb [hunk 1]
```diff
@@ -30,6 +30,9 @@
 :ssl_ca_file: <%= scope.lookupvar("foreman::client_ssl_ca") %>
 :ssl_priv_key: <%= scope.lookupvar("foreman::client_ssl_key") %>
 
+# HSTS setting
+:disable_hsts: <%= scope.lookupvar("foreman::disable_hsts") %>
```
Comment (ekohl) [inline templates/settings.yaml.erb]: IMHO this is a badly named variable: it should always be positive (so ```enable_hsts`` defaulting to ```true```)

Comment (tbrisker) [reply templates/settings.yaml.erb]: Still hasn't been merged in core - see related PR (and request change there so noone merges it)

Comment (tbrisker) [reply templates/settings.yaml.erb]: I thought about this and I don't think we should change it. If we default to true here and make the name e.g. `enable_hsts` that will cause any users who haven't updated their settings.yaml with the new option to default to false, thus disabling HSTS for them.

**Comment by tbrisker** [comment]
@ekohl updated to match the change in core

**Comment by tbrisker** [comment]
@ekohl core has been merged, rebased this on master.

File: manifests/init.pp [hunk 2]
```diff
@@ -206,6 +206,9 @@
 # $jobs_service::               Name of the service for running the background Dynflow executor
 #
 # $dynflow_in_core::            Whether the Dynflow executor service is provided by Foreman or tasks
+#
+# $hsts_enabled::               Disable HSTS enforcement in https requests
```
Comment (ekohl) [inline manifests/init.pp]: Disable -> Enable in the doc?

Comment (tbrisker) [reply manifests/init.pp]: done, thanks.

**Comment by mmoll** [comment]
merged, thanks @tbrisker!

**Pull request merged**